phoenix-core/src/main/java/org/apache/phoenix/iterate/BaseResultIterators.java

```diff
@@ -496,6 +496,10 @@ private static void optimizeProjection(StatementContext context, Scan scan, PTab
                         trackedColumnsBitset.set(qualifier);
                     }
                 }
+            } else {
+                // cannot use EncodedQualifiersColumnProjectionFilter in this case
+                // since there's an unknown set of qualifiers (cf.*)
+                trackedColumnsBitset = null;
```

Comment:
Yeah I think this makes sense, and hopefully should work for multiple filters too. Shall we add one more test to cover multiple cols and filters? Something like
`SELECT w.*, x.* FROM tableName WHERE y.v2='xyz' and z.v3='abc'` ?

Reply:
Yeah... Lemme add that test and then merge.
Thanks @virajjasani